File: VB/WindowsApplication3/MyRibbonPageHeaderViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports System.Drawing

Namespace DXSample
	Public Class MyRibbonPageHeaderViewInfo
		Inherits RibbonPageHeaderViewInfo
		Public Sub New(ByVal viewInfo As RibbonViewInfo)
			MyBase.New(viewInfo)
		End Sub

		Protected Overrides Function CalcActualHeaderRect(ByVal availableHeaderRect As Rectangle, ByVal bestSize As Size) As Rectangle
			Dim rect As Rectangle = MyBase.CalcActualHeaderRect(availableHeaderRect, bestSize)
			rect.Y += Ribbon.CustomSpace
			Return rect
		End Function

		Public Overrides Sub CalcPageHeader()
			MyBase.CalcPageHeader()
			Dim rect As Rectangle = Bounds
			rect.Y += Ribbon.CustomSpace
			Bounds = rect
		End Sub

		Protected Overrides Sub CalcHeaderItemsLayout(ByVal rect As Rectangle)
			rect.Y += Ribbon.CustomSpace
			MyBase.CalcHeaderItemsLayout(rect)
		End Sub

		Protected Shadows ReadOnly Property Ribbon() As MyRibbonControl
			Get
				Return TryCast(MyBase.Ribbon, MyRibbonControl)
			End Get
		End Property
	End Class
End Namespace

File: CS/WindowsApplication3/MyRibbonPageHeaderViewInfo.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using System.Drawing;

namespace DXSample {
    public class MyRibbonPageHeaderViewInfo : RibbonPageHeaderViewInfo
    {
        public MyRibbonPageHeaderViewInfo(RibbonViewInfo viewInfo) : base(viewInfo) { }

        protected override Rectangle CalcActualHeaderRect(Rectangle availableHeaderRect, Size bestSize)
        {
            Rectangle rect = base.CalcActualHeaderRect(availableHeaderRect, bestSize);
            rect.Y += Ribbon.CustomSpace;
            return rect;
        }

        public override void CalcPageHeader()
        {
            base.CalcPageHeader();
            Rectangle rect = Bounds;
            rect.Y += Ribbon.CustomSpace;
            Bounds = rect;
        }

        protected override void CalcHeaderItemsLayout(Rectangle rect)
        {
            rect.Y += Ribbon.CustomSpace;
            base.CalcHeaderItemsLayout(rect);
        }

        protected new MyRibbonControl Ribbon
        {
            get { return base.Ribbon as MyRibbonControl; }
        }
    }
}